test/unit/puppetlabs/services/legacy_routes/legacy_routes_core_test.clj

```diff
@@ -0,0 +1,145 @@
+(ns puppetlabs.services.legacy-routes.legacy-routes-core-test
+  (:require [clojure.test :refer :all]
+            [schema.test :as schema-test]
+            [puppetlabs.services.legacy-routes.legacy-routes-core :refer :all]
+            [puppetlabs.services.master.master-core :as master-core]
+            [ring.mock.request :as mock]
+            [ring.util.codec :as ring-codec]
+            [clojure.string :as str]))
+
+(use-fixtures :once schema-test/validate-schemas)
+
+(def master-mount "/puppet")
+(def ca-mount "/puppet-ca")
+(def master-api-version master-core/puppet-API-versions)
+(def ca-api-version master-core/puppet-ca-API-versions)
+(def accept-header [:headers "accept"])
+(def content-type-header [:headers "content-type"])
+
+(def accept-header-common-examples
+  {"raw" "binary"
+   "foo" "foo"
+   "foo,bar" "foo, bar"
+   "raw, foo" "binary, foo"
+   "foo, raw" "foo, binary"
+   "foo, raw, bar" "foo, binary, bar"})
+
+(def accept-header-raw-examples
+  (assoc accept-header-common-examples
+    "s, pson, raw" "s, pson, binary"))
+
+(def accept-header-s-pson-examples
+  (assoc accept-header-common-examples
+    "s, pson, raw" "binary"
+    "s, pson, foo" "binary, foo"))
+
+; [path [method [example expected]]]
+(def accept-header-test-data
+  {
+   "/" {:head {"raw" "raw"}
+        :post {"raw" "raw"}
+        :get {"raw" "raw"
+              "foo,bar" "foo,bar"}}
+   "/production/catalog/anode" {:get accept-header-raw-examples
+                                :post accept-header-raw-examples}
+   "/production/file_content/something" {:get accept-header-raw-examples}
+   "/production/file_bucket_file/something" {:get accept-header-s-pson-examples
+                                             :head accept-header-s-pson-examples
+                                             :put accept-header-s-pson-examples}
+   })
+
+(defn assertion-wrapper
+  "ring middleware that asserts by passing the request as the first argument
+  to assertion"
+  [handler assertion]
+  (fn [req]
+    (assertion req)
+    (handler req)))
+
+(defn validating-app
+  "Return a ring app with the results of build-ring-handler wrapped around a an
+  assertion-wrapper middleware as the master-handler.  The identity function is
+  the ca-handler."
+  [assertion]
+  (let [handler (assertion-wrapper identity assertion)]
+    (build-ring-handler
+      handler master-mount master-api-version
+      identity ca-mount ca-api-version)))
+
+(defn test-munged-request
+  "Validate a request by threading it through the legacy-routes request handler
+  into an assertion-wrapper ring middleware."
+  [request assertion]
+  ((validating-app assertion) request))
+
+(deftest test-legacy-routes
```

Comment:
We should have at least some coverage of cases where we're making a request that has some query parameters in the original URL to see that the `:environment` one is added to the list rather than just replacing them entirely.